### PR TITLE
Docs: Replace apt-key command with signed-by option

### LINKED ISSUE #35467: Docs: apt-key is deprecated and should not be used
https://github.com/grafana/grafana/blob/main/docs/sources/installation/debian.md#to-install-the-latest-enterprise-edition
> wget -q -O - https://packages.grafana.com/gpg.key | sudo apt-key add -

https://manpages.debian.org/buster/apt/apt-key.8.en.html#COMMANDS

> Note: Instead of using this command a keyring should be placed directly in the /etc/apt/trusted.gpg.d/ directory with a descriptive name and either "gpg" or "asc" as file extension.

Moreover, instead of `/etc/apt/trusted.gpg.d/` it should be placed in `/usr/share/keyrings/` with `signed-by` in sources.list.
https://wiki.debian.org/DebianRepository/UseThirdParty#Sources.list_entry. For example:
```
deb [signed-by=/usr/share/keyrings/deriv-archive-keyring.gpg] https://deriv.example.net/debian/ stable main
```

### PR DESCRIPTION
**Pull request by zuchka**
**What this PR does / why we need it**:

fixes #35467

replaces `apt-key` in our docs, which is deprecated, with the new best practices for 3rd party keys.

- https://wiki.debian.org/DebianRepository/UseThirdParty
- https://askubuntu.com/questions/1286545/what-commands-exactly-should-replace-the-deprecated-apt-key
- https://unix.stackexchange.com/questions/332672/how-to-add-a-third-party-repo-and-key-in-debian/582853#582853

Tested this on fresh GCP VMs running Ubuntu-20.04 LTS, 18.04 LTS, Debian 10, & Debian 11

**Which issue(s) this PR fixes**:

fixes #35467

Fixes #

**Special notes for your reviewer**:
fwiw `apt-key` is also used in three build scripts:

```
scripts/build/ci-e2e/Dockerfile
6:RUN curl -fsSL https://dl.google.com/linux/linux_signing_key.pub | apt-key add - && \

scripts/build/ci-build/Dockerfile
152:    curl -fsS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \

scripts/verify-repo-update/Dockerfile.deb
11:    curl https://packages.grafana.com/gpg.key | apt-key add -
```

### CODE REVIEW

File: docs/sources/installation/debian.md [hunk 1]
```diff
@@ -38,19 +38,19 @@ If you install from the APT repository, then Grafana is automatically updated ev
...
 sudo apt-get install -y software-properties-common wget
-wget -q -O - https://packages.grafana.com/gpg.key | sudo apt-key add -
+sudo wget -q -O /usr/share/keyrings/grafana.key https://packages.grafana.com/gpg.key
 ```
 
 Add this repository for stable releases:
 
 ```bash
-echo "deb https://packages.grafana.com/enterprise/deb stable main" | sudo tee -a /etc/apt/sources.list.d/grafana.list
+echo "deb [signed-by=/usr/share/keyrings/grafana.key] https://packages.grafana.com/enterprise/deb stable main" | sudo tee -a /etc/apt/sources.list.d/grafana.list
```
Comment (simondeziel) [inline docs/sources/installation/debian.md:47]: `s/\.key/.gpg/`

File: docs/sources/installation/debian.md [hunk 2]
```diff
@@ -38,19 +38,19 @@ If you install from the APT repository, then Grafana is automatically updated ev
 ```bash
 sudo apt-get install -y apt-transport-https
 sudo apt-get install -y software-properties-common wget
-wget -q -O - https://packages.grafana.com/gpg.key | sudo apt-key add -
+sudo wget -q -O /usr/share/keyrings/grafana.key https://packages.grafana.com/gpg.key
```
[the remark this answers is not in the document]
Comment (zuchka) [reply docs/sources/installation/debian.md]: hi @simondeziel, thanks for this feedback!

tested the updated version and it worked great. Thanks for getting this closer to Debian best practices 👍

**Comment by achatterjee-grafana** [comment]
Not stale

**Review by achatterjee-grafana — APPROVED** [review]

**Comment by simondeziel** [comment]
Not stale

**Comment by torkelo** [comment]
The docs file got moved to setup/installation, can you update the PR? We can then merge it

**Review by torkelo — APPROVED** [review]

**Comment by gelicia** [comment]
Got hit up about this on Twitter, is there an issue with this PR? https://twitter.com/motionthings/status/1548023893432029185

**Review by chri2547 — APPROVED** [review]

**Comment by zuchka** [comment]
@gelicia merged!  👍

**Pull request merged**